fast-api/mini-project/venv/src/main.py

```
@@ -0,0 +1,69 @@
+@app.post("/books", response_model=Book)
+def create_book(book: Book):
+    if any(b["id"] == book.id for b in books_db):
+        raise HTTPException(status_code=400, detail="Book with this ID already exists")
+    books_db.append(book.model_dump())
+    return book
+
+@app.put("/books/{id}", response_model=Book)
+def update_book(id: int, update_book: Book):
+    for idx, book in enumerate(books_db):
+        if book["id"] == id:
+            books_db[idx] == update_book.model_dump()
```

Ruff (0.12.2)

Pointless comparison. Did you mean to assign a value? Otherwise, prepend `assert` or remove it.

(B015)